File: WordPress/src/androidTest/java/org/wordpress/android/database/WordPressDBTest.java
package org.wordpress.android.database;

import android.content.Context;
import android.test.InstrumentationTestCase;
import android.test.RenamingDelegatingContext;

public class WordPressDBTest extends InstrumentationTestCase {
    protected Context mTestContext;
    protected Context mTargetContext;

    @Override
    protected void setUp() {
        // Run tests in an isolated context
        mTargetContext = new RenamingDelegatingContext(getInstrumentation().getTargetContext(), "test_");
        mTestContext = getInstrumentation().getContext();
    }
}

File: WordPress/src/androidTest/java/org/wordpress/android/database/WordPressDBTest.kt
package org.wordpress.android.database

import android.content.Context
import android.test.InstrumentationTestCase
import android.test.RenamingDelegatingContext
import org.wordpress.android.WordPressDB

class WordPressDBTest : InstrumentationTestCase() {
    private lateinit var mTestContext: Context
    private lateinit var mTargetContext: Context

    override fun setUp() {
        // Run tests in an isolated context
        mTargetContext = RenamingDelegatingContext(instrumentation.targetContext, "test_")
        mTestContext = instrumentation.context
    }

    fun testDatabaseCreation() {
        // This should run all migrations, and fail only if an exception is raised.
        // For instance, when the SQL statement is invalid
        WordPressDB(mTargetContext)
    }
}
